### ISSUE TITLE
`s3.read_parquet` with `chunked=True` fails if one of the parquet files in the dataset is empty

### ISSUE THREAD

**Issue by markopy** [aws/aws-sdk-pandas#2394]
### Describe the bug

Using `s3.read_parquet(s3_path, dataset=True, chunked=True)` fails if one of the files in the dataset contains no rows:
```
Traceback (most recent call last):
  File "<snip>.py", line 395, in bar
    for chunk_df in wr.s3.read_parquet(s3_path, dataset=True, chunked=True):
  File "<snip>/lib/python3.11/site-packages/awswrangler/s3/_read_parquet.py", line 248, in _read_parquet_chunked
    batch_size=batch_size, columns=columns, use_threads=use_threads_flag, use_pandas_metadata=False
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "pyarrow/table.pxi", line 3972, in pyarrow.lib.Table.from_batches
ValueError: Must pass schema, or at least one RecordBatch
```

The issue appears to be in the following piece of code from `_read_parquet_chunked()`:
```
            chunks = pq_file.iter_batches(
                batch_size=batch_size, columns=columns, use_threads=use_threads_flag, use_pandas_metadata=False
            )
            table = _add_table_partitions(
                table=pa.Table.from_batches(chunks),
                path=path,
                path_root=path_root,
            )
```
`chunks`  immediately raises `StopIteration` without returning anything which causes `pa.Table.from_batches()` to fail with `ValueError: Must pass schema, or at least one RecordBatch`.

### How to Reproduce

Use `s3.read_parquet(dataset=True, chunked=True)` to read a parquet dataset where one of the parquet files is empty.

### OS

Linux

### Python version

3.11

### AWS SDK for pandas version

3.2.1

**Comment by markopy** [comment]
I'm using pyarrow 12.0.1. Since reading datasets with zero row files appears to have worked previously it's possible the breaking change happened in pyarrow.

See also https://github.com/apache/arrow/issues/35713 for the same error.

**Comment by kukushking** [comment]
HI @markopy fixed in #2400, will be available in the next release. Thanks!

**Comment by markopy** [comment]
Thanks for the quick fix @kukushking!
